### PR TITLE
NMS-8403: Document styles of identifying nodes in resource IDs

### PR DESCRIPTION
**Pull request by jeffgdotorg**
Also tweaked one example in Measurements API section to cut confusion.

* JIRA: http://issues.opennms.org/browse/NMS-8403

### CODE REVIEW

**Comment by jeffgdotorg** [comment]
Review for accuracy please

**Comment by j-white** [comment]
It's accurate. Should we elaborate on when to use node[] vs nodeSource[]?

**Comment by jeffgdotorg** [comment]
Altered the paragraph about choosing a style to reflect the reality that nodeSource[N] is always safe, and that this could simply many client implementations.

File: opennms-doc/guide-development/src/asciidoc/text/rest/rest-api.adoc [hunk 1]
```diff
@@ -79,3 +79,44 @@ The following apply for OpenNMS Horizon 18 and newer.
...
+|===
+
+The two styles are not interchangeable. For example, it is not valid to use `node[Routers:atledg04]` nor to use `nodeSource[175]`.
+
+IMPORTANT: On systems where `storeByForeignSource` is enabled, you must *always* use the `nodeSource` style when referring to requisitioned nodes and `node` when referring to auto-provisioned nodes. The following tables summarize which style is permissible in which situation. Auto-provisioned nodes are ones that do not belong to any requisition. As these nodes lack a foreign-source name, they intrinsically cannot be represented using `nodeSource` style identifiers.
+
+.If `storeByForeignSource` is `false`
+[options="header"]
+|===
+| Node Provisioning Style | `node[]` Valid? | `nodeSource[]` Valid?
```
Comment (indigo423) [inline opennms-doc/guide-development/src/asciidoc/text/rest/rest-api.adoc]: Hidden question in the text here, seems to need clarification.

File: opennms-doc/guide-development/src/asciidoc/text/rest/rest-api.adoc [hunk 2]
```diff
@@ -79,3 +79,44 @@ The following apply for OpenNMS Horizon 18 and newer.
...
+| Requisitioned           | Yes             | Yes
+| Auto-Provisioned        | Yes             | No
+|===
+
+.If `storeByForeignSource` is `true`
+[options="header"]
+|===
+| Node Provisioning Style | `node[]` Valid? | `nodeSource[]` Valid?
```
Comment (indigo423) [inline opennms-doc/guide-development/src/asciidoc/text/rest/rest-api.adoc]: Same here, Hidden question in the text here, seems to need clarification.

**Comment by jeffgdotorg** [comment]
@indigo423 I don't understand the significance of "hidden question'. Can you elaborate?

**Comment by indigo423** [comment]
The text contains something like "Valid?", it seems there is something which needs clarification? It's just easy to miss.

**Pull request merged**